### ISSUE TITLE
Arrangement_2 Memory leak, Class is not fully destructed

### ISSUE THREAD

**Issue by mu116699** [CGAL/cgal#7430]
Recently, when I was checking memory leaks, I found that simply constructing an Arrangement_2 variable and using deleaker to detect memory leaks. Is this due to the CGAL library itself or my compilation problem?

CGAL demo：
```c++
int main()
{

	//Point_2 p(1, 1), q(10, 10);

	////Point_2 m(5, 9);
	//Point_2 m(10, -6);
	//switch (CGAL::orientation(q,p,  m)) {
	//case CGAL::COLLINEAR://COLLINEAR 共线
	//	std::cout << "are collinear\n";
	//	break;
	//case CGAL::LEFT_TURN:
	//	std::cout << "make a left turn\n";
	//	break;
	//case CGAL::RIGHT_TURN:
	//	std::cout << "make a right turn\n";
	//	break;
	//}

	typedef CGAL::Exact_predicates_exact_constructions_kernel K;
	typedef CGAL::Arr_segment_traits_2<K> Traits;
	typedef Traits::Point_2 Point_2;
	typedef Traits::X_monotone_curve_2 Segment_2;
	typedef CGAL::Arrangement_2<Traits> Arrangement_2;
	Arrangement_2 *parr = nullptr;
	{
		Arrangement_2 arr;
		parr = &arr;
	}
	
	auto curve = Segment_2(Point_2(0, 0), Point_2(2, 2));
	//CGAL::insert(arr, curve);
	//arr.clear();
	
	return 0;
}
```

**Comment by lrineau** [comment]
> ```c++
> 	Arrangement_2 *parr = nullptr;
> 	{
> 		Arrangement_2 arr;
> 		parr = &arr;
> 	}
> ```

That code is wrong: after the exit of the scope, the local variable `arr` is destroyed, and then `parr` is a pointer to a destroyed object.

**Comment by mu116699** [comment]
Hello, I know this is wrong, I mainly simply use a pointer to point to the arr address, but found that the arr internal decl variable was not released after returning, I made a demonstration here, modify the representative like this,

typedef CGAL::Exact_predicates_exact_constructions_kernel K;
typedef CGAL::Arr_segment_traits_2<K> Traits;
typedef Traits::Point_2 Point_2;
typedef Traits::X_monotone_curve_2 Segment_2;
typedef CGAL::Arrangement_2<Traits> Arrangement_2;

int main()
{
	Arrangement_2 arr;
	return 0;
}
There is still a memory leak. It is that there are internal problems in Arrangement_2. I want to ask how to solve them, and I hope you can give me a solution. The CGAL version I use is 5.3.2. thank you.

**Comment by sloriot** [comment]
How did you detect the leak? I tried with address sanitizer and I did not get any error reported. I also put an infinite while loop around `Arrangement_2 arr;` and did not see any memory usage increase after a few minutes.

**Comment by sloriot** [comment]
Could you please copy/paste the error message?

**Comment by mu116699** [comment]
pool.hpp, line 97 (e:\cgal\boost_1_71_0\boost\pool\pool.hpp)	0x00942850	1	Heap memory	CGALdemo.exe	F:\CGALdemo\Debug\CGALdemo.exe	1708	9984	2023/5/10 18:09:05.6730000	213	35484
Hello, the above shows the leak shown in the program, but if it is replaced by your official version 5.5, there is no problem, but now we are using 5.3.2, we want to find out why this is caused and how to solve it. No matter what the result is, thank you very much for your hard work in these two days.

**Comment by sloriot** [comment]
I missed that you were using an old version of CGAL. This boost memory pool issue has been fixed in https://github.com/CGAL/cgal/pull/6638. Mimimal CGAL version with the fix is 5.4.2
